### ISSUE TITLE
Earlystopping 'auto' mode is garbage

### ISSUE THREAD

**Issue by qmarcou** [tensorflow/tensorflow#59036]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.1

### Custom Code

No

### Current Behaviour?

Trying to use the EarlyStopping callback, the documentation states for the 'auto' mode:
> mode: One of `{"auto", "min", "max"}`. In `min` mode,
        training will stop when the quantity
        monitored has stopped decreasing; in `"max"`
        mode it will stop when the quantity
        monitored has stopped increasing; in `"auto"`
        mode, the direction is automatically inferred
        from the name of the monitored quantity.

In fact the [tensorflow code](https://github.com/tensorflow/tensorflow/blob/cfc5ac84d80b705b8fc83350e7d5ca34e1daac17/tensorflow/python/keras/callbacks.py) is just:
```shell
if mode == 'min':
      self.monitor_op = np.less
    elif mode == 'max':
      self.monitor_op = np.greater
    else:
      if 'acc' in self.monitor:
        self.monitor_op = np.greater
      else:
        self.monitor_op = np.less
```

This means that for any metric that is basically not the accuracy the 'auto' mode will default to 'min', while a vast majority of performance metrics are designed to be increasing functions of performance.

This documentation is extremely misleading, and given this behavior I think we can safely say that almost nothing is inferred from the monitored metric.

In order to protect other users from this trap, I would advise to remove the 'auto' mode and its associated argument altogether.

### Standalone code to reproduce the issue

```shell
EarlyStopping(monitor='val_precision',
               mode='auto')
# this callback will tend to minimize precision...
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @qmarcou !
Thanks for sharing your observation .

Could you create a PR in  [keras-team/keras repo](https://github.com/keras-team/keras/issues) for review.
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . 

Thank you!
